usr/share/rear/layout/prepare/GNU/Linux/100_include_partition_code.sh

```diff
@@ -225,9 +227,15 @@ EOF
             if [[ "$end" ]] ; then
                 end=$( mathlib_calculate "$end - 1" )
             fi
-            cat >> "$LAYOUT_CODE" <<EOF
+            if [[ "$ARCH" == "Linux-s390" ]] ; then
+               # if dasd disk is LDL formated, then do not partition it, because it is partitioned and can take only partition
+               if [[ ! "${listDasdLdl[@]}" =~ "$device" ]] ; then
+                 echo "not LDL dasd formated disk, create a partition"
+                 cat >> "$LAYOUT_CODE" <<EOF
 create_disk_partition "$device" "$name" $number $start $end
```

Comment:
Now I see that this looks plain wrong because it changes
```
        if [[ "$FEATURE_PARTED_ANYUNIT" ]] ; then
            if [[ "$end" ]] ; then
                end=$( mathlib_calculate "$end - 1" )
            fi
            cat >> "$LAYOUT_CODE" <<EOF
create_disk_partition "$device" "$name" $number $start $end
EOF
        else
```
to
```
if [[ "$FEATURE_PARTED_ANYUNIT" ]] ; then
            if [[ "$end" ]] ; then
                end=$( mathlib_calculate "$end - 1" )
            fi
            if [[ "$ARCH" == "Linux-s390" ]] ; then
               # if dasd disk is LDL formated, then do not partition it, because it is partitioned and can take only partition
               if [[ ! "${listDasdLdl[@]}" =~ "$device" ]] ; then
                 echo "not LDL dasd formated disk, create a partition"
                 cat >> "$LAYOUT_CODE" <<EOF
create_disk_partition "$device" "$name" $number $start $end
EOF
               fi
            fi
        else
```
which means
```
create_disk_partition "$device" "$name" $number $start $end
```
appears in diskrestore.sh only
```
if [[ "$ARCH" == "Linux-s390" ]] ; then
```

Reply:
Fixed by
https://github.com/rear/rear/pull/2142/commits/398b0473f506d57d1e523c6ef14456be566bc52b